infrastructure/src/main/java/org/corfudb/infrastructure/log/Segment.java

```diff
@@ -529,6 +548,51 @@ public void close() {
         }
     }
 
+    /**
+     * This index maps a sequence number to a file offset + payload length.
+     *
+     * The file offset and payload length are packed in a long (treated as unsigned) as follows:
+     *
+     *  |--------64-bits--------|
+     *  |fileOffset|payloadSize|
+     *
+     */
+    private static final class Index {
+
+        private final BoundedMap map;;
+
+        private Index(long offset, int size) {
+            this.map = new BoundedMap(offset, size);
+            int highBitsNum = Long.bitCount(MAX_SEGMENT_SIZE);
```

Comment:
Since the bitCount return values are constant here, they can be stored so that the result does not have to be recomputed for every pack/unpackOffset.

Reply:
Would be surprised if the popcount method is not inlined either in the JIT phase, or the tiered compilation.

Anyways, moved to class static.